keras/layers/cudnn_recurrent.py

```diff
@@ -9,7 +9,7 @@
 from .. import regularizers
 from .. import constraints
 from .recurrent import RNN
-from ..layers import InputSpec
+from keras.engine.base_layer import InputSpec
```

Comment:
Use package-relative imports inside the Keras codebase, don't use `import keras` (only valid in unit tests)

Reply:
Done. The diff is much cleaner now.